pkgs/build-support/node/fetch-npm-deps/src/main.rs

```diff
@@ -35,10 +35,29 @@ struct OldPackage {
 
 #[derive(Debug, Deserialize, PartialEq, Eq)]
 struct Package {
-    resolved: Option<Url>,
+    resolved: Option<UrlOrString>,
     integrity: Option<String>,
 }
 
+#[derive(Debug, Deserialize, PartialEq, Eq)]
+#[serde(untagged)]
+enum UrlOrString {
+    Url(Url),
+    String(String),
+}
+
+impl From<Url> for UrlOrString {
+    fn from(u: Url) -> UrlOrString {
+        UrlOrString::Url(u)
+    }
+}
+
+impl From<String> for UrlOrString {
+    fn from(s: String) -> UrlOrString {
+        UrlOrString::String(s)
+    }
+}
```

Comment:
Nit: Are these `From` implementations necessary? In the code, it seems like `::from` is called only when we already know that it's transforming a `Url` to a `UrlOrString`, and I feel like doing so explicitly reads clearer in that case.